lib/tests/test_persist_state.c

```diff
@@ -206,6 +207,25 @@ Test(persist_state, test_persist_state_remove_entry)
   cancel_and_destroy_persist_state(state);
 }
 
+Test(persist_state, test_persist_state_string_alloc_test)
+{
+  guint8 version;
+  gsize size_1;
+  gsize size_2;
+
+  PersistState *state = clean_and_create_persist_state_for_test("test_persist_state_string_alloc.persist");
+
+  persist_state_alloc_string(state, "test", "test_longer", -1);
+  persist_state_lookup_entry(state, "test", &size_1, &version);
+
+  persist_state_alloc_string(state, "test", "test_short", -1);
+  persist_state_lookup_entry(state, "test", &size_2, &version);
+
+  cr_assert_eq(size_1, size_2, "allocation versions do not match");
+
+  cancel_and_destroy_persist_state(state);
+}
```

Comment:
Thanks for writing a test.

Can we also add a test case, where test "reuse" scenarios?
You could look up the entries with the non-string API and compare their `PersistEntryHandle` to check whether they are identical.

Reply:
Good idea, I'll see what I can do.